### ISSUE TITLE
arel 4.0.1 method signature change breakage

### ISSUE THREAD

**Issue by seanwalbran** [smathy/wherex#3]
This commit in arel 4.0.1, used by rails 4.0.1 which was just released, changes the signatures of the private methods that wherex is hooking into to work: https://github.com/rails/arel/commit/68a95542e1a7a79d9777223fbffd2b982fed0268

The impact is pretty severe, in that it affects queries that don't even try to use the wherex functionality, e.g.:

```
[3] pry(main)> User.where(:id => 1).first
ArgumentError: wrong number of arguments (2 for 1)
from /Users/seanwalbran/.rvm/gems/ruby-1.9.3-p125@socialcast/gems/wherex-1.0.8/lib/wherex/visitor.rb:11:in `visit_Arel_Nodes_Equality_with_wherex'
```

It looks to me like this kind of fix would work, but in a rails-version-4.0.1-or-above-only kind of way: https://gist.github.com/seanwalbran/7272750

**Comment by javierg** [comment]
that will work, just set a default value on a to nil,for backwards compatibility

```
def visit_Arel_Nodes_Equality_with_wherex o, a = nil
```

**Comment by javierg** [comment]
@smathy I have a patch, woth this change, If you give me access I push the PR.

**Comment by smathy** [comment]
Done.

**Comment by smathy** [comment]
Pushed this in v1.1.0 just now with all the Rails 4 compatibility changes.

Thanks @javierg for your fixes in 40ad6615bd21f8d7f8d88930ef572922337f9a93, I made them work in both Rails4 and 3 with 5a8c93f1
